plugins/modules/mso_schema_site_anp_epg_bulk_staticport.py

```diff
@@ -374,95 +375,89 @@ def main():
-
-            if state == "query":
-                if leaf is None or vlan is None:
-                    mso.existing = mso_objects.get("site_anp_epg").details.get("staticPorts")
-                elif not mso.existing:
-                    mso.fail_json(msg="Static port '{portpath}' not found".format(portpath=portpath))
-                mso.exit_json()
-
-            new_leaf = dict(
-                deploymentImmediacy=deployment_immediacy,
-                mode=mode,
-                path=portpath,
-                portEncapVlan=vlan,
-                type=path_type,
-            )
-
-            if primary_micro_segment_vlan:
-                new_leaf.update(microSegVlan=primary_micro_segment_vlan)
-
-            # validate and append staticports to staticport_list if path variable is different
-            if path in unique_paths:
-                mso.fail_json(msg="Path should be unique for each Static Port")
-            else:
-                unique_paths.append(path)
-                staticport_list.append(new_leaf)
-
-    # If payload is empty, anp and EPG already exist at site level
-    if not payload:
-        op_path = ports_path
-        payload = staticport_list
-    else:
-        # If anp already exists at site level
-        if "anpRef" not in payload:
-            payload["staticPorts"] = staticport_list
-        else:
-            payload["epgs"][0]["staticPorts"] = staticport_list
-
     mso.previous = mso.existing
 
     if state == "absent":
         if mso.existing:
-            mso.sent = mso.existing = {}
-            ops.append(dict(op="remove", path=ports_path))
+            mso.sent = mso.existing = []
+            ops.append(dict(op="remove", path=op_path))
 
     elif state == "present":
-        mso.sanitize(payload, collate=True)
+        if static_ports is not None:
+            for static_port in static_ports:
+                path_type = static_port.get("type") or module_path_type
+                pod = static_port.get("pod") or module_pod
+                leaf = static_port.get("leaf") or module_leaf
+                fex = static_port.get("fex") or module_fex
+                path = static_port.get("path") or module_path  # Note :path has to be diffent in each leaf for every static port in the list.
+                vlan = static_port.get("vlan") or module_vlan
+                primary_micro_segment_vlan = static_port.get("primary_micro_segment_vlan") or module_primary_micro_segment_vlan
+                deployment_immediacy = static_port.get("deployment_immediacy") or module_deployment_immediacy
+                mode = static_port.get("mode") or module_mode
+
+                required_dict = {"pod": pod, "leaf": leaf, "path": path, "vlan": vlan}
+                if None in required_dict.values():
+                    res = []
+                    for i in required_dict:
+                        if required_dict[i] is None:
+                            res.append(i)
```

Comment:
not needed but you could use list comprehension here
```suggestion
                    res = [ key for key in required_dict.keys() if required_dict[key] is None ]
```